lib/teleterm/services/clientcache/clientcache.go

```diff
@@ -79,9 +79,16 @@ func (c *Cache) Get(ctx context.Context, clusterURI uri.ResourceURI) (*client.Pr
 			return nil, trace.Wrap(err)
 		}
 
-		//nolint:staticcheck // SA1019. TODO(gzdunek): Update to use client.ClusterClient.
-		newProxyClient, err := clusterClient.ConnectToProxy(ctx)
-		if err != nil {
+		var newProxyClient *client.ProxyClient
+		if err := clusters.AddMetadataToRetryableError(ctx, func() error {
+			//nolint:staticcheck // SA1019. TODO(gzdunek): Update to use client.ClusterClient.
+			proxyClient, err := clusterClient.ConnectToProxy(ctx)
```

Comment:
```suggestion
			newProxyClient, err = clusterClient.ConnectToProxy(ctx)
```
can we do this instead?

Reply:
It would work too, but idk, I feel more safe creating a new variable than overwriting a one from the outer scope. Unfortunately, it's quite verbose here.